cli-plugins/plugin/plugin.go

```diff
@@ -53,7 +53,12 @@ func RunPlugin(dockerCli *command.DockerCli, plugin *cobra.Command, meta manager
 
 // Run is the top-level entry point to the CLI plugin framework. It should be called from your plugin's `main()` function.
 func Run(makeCmd func(command.Cli) *cobra.Command, meta manager.Metadata) {
-	dockerCli, err := command.NewDockerCli()
+	var dockerCliOpts []command.DockerCliOption
+	if ua, ok := userAgent(meta); ok {
+		dockerCliOpts = append(dockerCliOpts, command.WithUserAgent(ua))
+	}
+
+	dockerCli, err := command.NewDockerCli(dockerCliOpts...)
```

Comment:
Sorry for the delay (again), this one dropped off my list.

I was just wondering; do we actually need the Metadata to have the new field? Would it also work if the plugin would use `dockerCli.Apply()` and set the user agent? 

Looking at compose; https://github.com/docker/compose/blob/e5cd265abbdd9d67d76558d3b8d2e7fec68b7573/cmd/main.go#L36-L39

It gets the dockerCLI passed; which (with the option you added in this PR) could use that function to set a custom user-agent;

```go
func pluginMain() {
	plugin.Run(func(dockerCli command.Cli) *cobra.Command {
		_ = dockerCli.Apply(command.WithUserAgent("whatever"))
```

The net-result would be the same, but without the `Metadata` as intermediate (because that one also gets used for the JSON response, where it's not used?)